src/modes/shop/shop.cpp

```diff
@@ -1902,92 +1902,100 @@ void ShopMode::SetPriceLevels(SHOP_PRICE_LEVEL buy_level,
     _sell_price_level = sell_level;
 }
 
-void ShopMode::AddItem(uint32_t object_id, uint32_t stock)
+bool ShopMode::AddItem(uint32_t object_id, uint32_t stock)
 {
     if(IsInitialized()) {
         IF_PRINT_WARNING(SHOP_DEBUG) << "Function called after shop was already initialized."
                                      << std::endl;
-        return;
+        return false;
     }
 
     if(object_id == private_global::OBJECT_ID_INVALID ||
-           object_id >= private_global::OBJECT_ID_EXCEEDS) {
+       object_id >= private_global::OBJECT_ID_EXCEEDS) {
         PRINT_WARNING << "attempted to add object with invalid id: "
                       << object_id << std::endl;
-        return;
+        return false;
     }
 
     if(_available_buy.find(object_id) != _available_buy.end()) {
         PRINT_WARNING << "attempted to add object that already existed: "
                       << object_id << std::endl;
-        return;
+        return false;
     }
 
     std::shared_ptr<GlobalObject> new_object =
-        GlobalCreateNewObject(object_id, 1);
-    if (new_object != nullptr) {
-        ShopObject *new_shop_object = new ShopObject(new_object);
-        if (stock > 0)
-            new_shop_object->IncrementStockCount(stock);
-        else // When the stock is set to 0, it means there is an infinity amount of object to buy.
-            new_shop_object->SetInfiniteAmount(true);
-        _available_buy.insert(std::make_pair(object_id, new_shop_object));
-    }
+      GlobalCreateNewObject(object_id, 1);
+    if (new_object == nullptr) {
+        return false;
+    }
+
+    ShopObject *new_shop_object = new ShopObject(new_object);
```

Comment:
Since you're changing, could you put the star next to the type and not the variable name? E.g: ShopObject* new_shop_object